code/datums/action.dm

```diff
@@ -500,6 +500,16 @@
 		return spell.can_cast(owner)
 	return 0
 
+/datum/action/spell_action/UpdateButtonIcon()
+	if(button && !..())
+		var/obj/effect/proc_holder/spell/S = target
+		if(!istype(S))
+			return
+		var/progress = S.get_availability_percentage()
+		var/col_val_high = 72 * progress + 128
+		var/col_val_low = 200 * progress
+		button.color = rgb(col_val_high, col_val_low, col_val_low, col_val_high)
```

Comment:
I'm a bit nervous that there's not a clear delineation between "99% recharged" and "ready"

Could there be some kind of skip of the color level at 100%?

Reply:
Or, wait - the gif you provide seems to imply that it does
let's see, `progress` is a value between `0` and `1`
at 0, `col_val_high = 128` and `col_val_low = 0`
at 1, `col_val_high = 200` and `col_val_low = 200`
What about this would cause the distinct color change between 0.99 completion and 1.0 completion...?

Reply:
The way it's currently implemented will make the color shift from about 78% of the full color to 100% of the sprite. I'll see about making this more clear though